packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -228,6 +248,34 @@ class CreateCommand extends FlutterCommand {
     }
   }
 
+  _ProjectType getProjectType(Directory projectDir) {
+    _ProjectType template;
+    _ProjectType detectedProjectType;
+    final bool metadataExists = projectDir.absolute.childFile('.metadata').existsSync();
+    if (argResults['template'] != null) {
+      template = _stringToProjectType(argResults['template']);
+    } else {
+      if (projectDir.existsSync() && projectDir.listSync().isNotEmpty) {
+        detectedProjectType = _determineTemplateType(projectDir);
+        if (detectedProjectType == null && metadataExists) {
+          // We can only be definitive that this is the wrong type if the .metadata file
+          // exists and contains a type that we don't understand, or doesn't contain a type.
+          throwToolExit('Sorry, unable to detect the type of project to recreate. '
```

Comment:
I assume runCommand will always crash first? We wouldn't want the usageValues getter to crash.

Reply:
That is correct. The tool will exit at that point, so we won't get stats.